src/utils/core/hooks/effect.ts

```diff
@@ -0,0 +1,72 @@
+import { global } from "@/utils/core/registry/hooks";
+import { updateSchedule } from "@/utils/core/hooks/scheduler";
+
+type TCleanupFunction = () => void;
+type TEffectCallback = () => void | TCleanupFunction;
+type TDependencyList = any[] | null;
+
+export interface IEffect {
+  callback: TEffectCallback;
+  deps: TDependencyList;
+  cleanup?: TCleanupFunction;
+}
+
+const shouldRerunEffect = (
+  prevDeps: TDependencyList,
+  nextDeps: TDependencyList
```

Comment:
P5;

소소한 생각이지만, TDependencyList에 null 타입도 추가하는 것이 정말 좋은 구조인지 의문이 있어요.

TDependencyList가 null도 들어올 수 있게 함으로 TDependencyList type을 쓰는 곳에선 항상 값이 있음을 신뢰하기 어렵기 때문이에요. (항상 null에 대한 핸들링을 해줘야함)

그렇기 때문에 꼭 null을 처리해야하는 경우가 아니라면 TDependencyList는 [] 타입만 남겨두고 아래처럼 옵셔널 함을 추가하는게 더 좋지 않을까 생각해요.

 ```suggestion
  prevDeps?: TDependencyList,
  nextDeps?: TDependencyList
```

Reply:
더 나아가 이렇게 처리하면 아래 null 체크도 간결하게 가져갈 수 있어요!

```suggestion
  prevDeps?: TDependencyList = [],
  nextDeps?: TDependencyList = []
```

Reply:
첨언해주신 의견은 유즈이펙트의 인자가 null일때(이펙트를 조건없이 리렌더 마다 실행해야 하는 케이스)를 처리할 수 없어서 리액트와 동일한 구현을 할 수 없게 될것 같아요!
List가 빈배열이 아니고 null이 될 수 있다는건 개념적으로 이상한게 이해되므로 좀더 의미에 맞게 수정해보겠습니다.